### ISSUE TITLE
Followers Unable to become leader with aws-pca consul/1.7.2 or 1.7.1 | Could not Find Region Configuration when generating CA root certificate

### ISSUE THREAD

**Issue by dmore** [hashicorp/consul#7529]
#### Overview of the Issue

When using connect with aws-pca when leader is gone by mistake the new instances are unable to lead. Problem seems to be due to Missing Region.

  agent.server: cluster leadership acquired
2020-03-27T15:41:46.828Z [INFO]  agent.server: New leader elected: payload=ip-10-70-x-a
2020-03-27T15:41:46.829Z [INFO]  agent.server.raft: pipelining replication: peer="{Voter 4f8c4b6f-51d9-71a7-015b-6dc5dac91837 10.70.x.b:8300}"
2020-03-27T15:41:46.869Z [INFO]  agent.server: initializing acls
2020-03-27T15:41:46.869Z [INFO]  agent.leader: started routine: routine="legacy ACL token upgrade"
2020-03-27T15:41:46.869Z [INFO]  agent.leader: started routine: routine="acl token reaping"
2020-03-27T15:41:46.869Z **[ERROR] agent.server: failed to establish leadership: error="error generating CA root certificate: MissingRegion: could not find region configuration"**
2020-03-27T15:41:46.869Z [INFO]  agent.server.raft: pipelining replication: peer="{Voter 27fc7cc4-5477-a8c2-8c46-5bdb6c904f9b 10.70.x.c:8300}"
2020-03-27T15:41:46.883Z [INFO]  agent.server: successfully transferred leadership: attempt=0 retry_limit=3
2020-03-27T15:41:46.905Z [INFO]  agent.server.raft: entering follower state: follower="Node at 10.70.x.a:8300 [Follower]" leader=
2020-03-27T15:41:46.905Z [INFO]  agent.server.raft: aborting pipeline replication: peer="{Voter 27fc7cc4-5477-a8c2-8c46-5bdb6c904f9b 10.70.x.c:8300}"
2020-03-27T15:41:46.905Z [INFO]  agent.server.raft: aborting pipeline replication: peer="{Voter 4f8c4b6f-51d9-71a7-015b-6dc5dac91837 10.70.x.b:8300}"
2020-03-27T15:41:46.905Z [INFO]  agent.server: cluster leadership lost
2020-03-27T15:41:46.958Z [INFO]  agent.server: New leader elected: payload=ip-10-70-x-c
2020-03-27T15:41:46.958Z [INFO]  agent.server: New leader elected: payload=ip-10-70-x-c

#### Reproduction Steps

Terminate the leader. Add several followers. Followers unable to lead.

Steps to reproduce this issue, eg:
(...)
 "connect": {
    "enabled" : true,
    "ca_provider" : "aws-pca",
    "ca_config" : {
      "existing_arn" : "arn:aws:acm-pca:region_far:user:certificate-authority/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
    }
  },
(...)

setting this server config with root_arn and adding a region like below does not acknowledge either. Always getting that region error.

ca_config
   region: x
   root_arn: arn......../certificate-authority....

aws acm-pca list-certificate-authorities --max-results 10 works. that is aws cli version 2.

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
output from client 'consul info' command here
```

</details>

<details>
  <summary>Could not Find Region Configuration when generating CA root certificate</summary>

```
output from server 'consul info' command here
```

</details>

### Operating system and Environment details

OS, Architecture, and any other information you can provide about the environment.

### Log Fragments

Include appropriate Client or Server log fragments. If the log is longer than a few dozen lines, please include the URL to the [gist](https://gist.github.com/) of the log instead of posting it in the issue. Use `-log-level=TRACE` on the client and server to capture the maximum log detail.

**Comment by mkeeler** [comment]
This doesn't just affect PCA but any external CA provider (Vault, PCA ...). If we cannot reach them when establishing leadership then it is failing causing infinite leader elections.

**Comment by blake** [comment]
Hi @dmore, this issue was fixed in Consul 1.9.2 by PR #9570. I'm going to close this issue. Please comment or open another issue if this is still a problem.
